sentry_sdk/integrations/aws_lambda.py

```diff
@@ -234,16 +233,29 @@ def _make_request_event_processor(aws_event, aws_context, configured_timeout):
 
     def event_processor(event, hint, start_time=start_time):
         # type: (Event, Hint, datetime) -> Optional[Event]
-        extra = event.setdefault("extra", {})
+        total_memory, used_memory, free_memory = map(
```

Comment:
@rhcarvalho @untitaker This calculation of 'used_memory' is incorrect, but it is the only source I could find up until now.
I've done some manual testing on this, it seems to be in alignment with memory usage displayed in AWS Lambda console, but I couldn't see an exact formula which calculates the desired result.

Reply:
Not a big fan of this. It looks at the memory consumption for the whole system, not a single process.
Please consider alternatives like https://stackoverflow.com/a/7669482/963287

Would be good to have some error handling here to avoid blowing everything up if we can't assess memory usage (graceful degradation, progressive enhancement).

Please update the PR title/description if you're changing the scope of this PR. IIRC memory usage was not part of the original patch.

Reply:
@rhcarvalho I've tried with the above mentioned method, it gave result as 22 MB in case where max memory usage was 61 MB. So, still wrong results. But, I've checked 'resource' library and it has some more parameters for different memory usage. I'll check those once and see if they add up to the exact value which is returned in the AWS lambda result console.

Reply:
👍 

Note that some memory is allocated after this code runs, so it wouldn't be a big surprise to see different numbers reported here and by the lambda runtime, though the difference shouldn't be significant (otherwise we have a performance improvement to go after in the SDK).

Reply:
I had multiple runs of the code and the difference is actually this significant only.
I went through the link : https://docs.python.org/3/library/resource.html, I saw other parameters related to memory usage like ru_ixrss (shared memory size), ru_idrss (unshared memory size) & ru_isrss (unshared stack size), these parameters showed memory usage as 0.

The memory usage which comes as output is 24680 (in kilobytes) or a value around this. And the value which is displayed on AWS lambda console is 62 MB.

So, shall we keep using this value itself ? Also, wanted to know abut the graceful degradation suggestion on the previous comment.

Reply:
By graceful degradation I mean we should have code written with error handling in mind, and isolate the errors as much as possible. E.g. if there is an exception when trying to figure out the memory usage, the outcome should be that we do not report on memory usage, while reporting on other aspects is unaffected. And particularly, an exception in any part of collecting extra event data should never prevent the SDK from sending the event.

Reply:
@rhcarvalho will be removing memory usage data, so resolving this issue.